### PR TITLE
Add Circular Restricted Three Body Problem Package

### PR DESCRIPTION
**Pull request by JackCrusoe47**
# Added a CR3BP library with basic functionality. 

I added a CR3BP function set with basic propagation functionality. The code is in a folder since I plan to add additional functionality soon. I am still a noob with Python, hence most of my code are direct translations from my MATLAB package. I still have not integrated **astropy** constants and units with my code and currently, I use **matplotlib** and **mplot3d** for plotting. I require help with the integration of poliastro plotting environment.

**Note**: all computations currently are performed in non-dimensional units for numerical stability. All parameters are non-dimensionalized with characteristic values (length, mass, time, velocity and angular velocity).

## Current functionality
- propagation of orbits in CR3BP 
- propagation with State-Transition-Matrix
- compute characteristic values of CR3BP given primary body k values and mean distance
- test file to check implementation with a few real orbits (plots are in the non-dimensional rotating frame)
    - Lunar Gateway Station NRHO orbit (9:2 L2 Southern NRHO)
    - Distant Retrograde Orbit
    - Butterfly Orbit
    - Vertical Lyapanov Orbit

## Future Plans
- differential corrector using STM
- compute periodic orbit families
- eigenvalue/eigenvector calculation for Monodromy matrix
- frame conversions (dimensional inertial <-> non-dimensional rotating)
- analytical approximate solutions for famous orbit families in CR3BP

I plan to implement most of the functionality present in https://github.com/JackCrusoe47/CR3BP_MATLAB_Library

Do let me know of any code improvements, computation errors or other elements you find.

cheers,

Kevin

### CODE REVIEW

**Comment by JackCrusoe47** [comment]
4 Orbit examples in test file:
Figure_1
Figure_2
Figure_3
Figure_4

File: contrib/CR3BP/CR3BP.py
Comment (jorgepiloto) [inline contrib/CR3BP/CR3BP.py:1]: May you please provide a little description or links to the references you used when implementing the different equations within this script?

File: contrib/CR3BP/test_run_CR3BP.py
Comment (jorgepiloto) [inline contrib/CR3BP/test_run_CR3BP.py:148]: In order to get all the figures rendered in the screen, a `plt.show()` statement must be added.

**Review by jorgepiloto — CHANGES_REQUESTED** [review]
Thanks for your contribution @JackCrusoe47. I left some comments and requested changes in the script but it looks great! 🚀

Before this can be merged within the `contrib/` directory, the following task needs to be solved:

* The quality tests are not passing. Try to execute `tox -e reformat && tox -e check`. The first command will apply `black` and `isort` styles to the code you just created while the second command will run the quality checks again. Once the `tox -e check` no longer raises an error, the quality checks are ready.

Finally, pinging here @astrojuanlu  so he can give you feedback when possible 👍🏽

**Comment by JackCrusoe47** [comment]
@jorgepiloto Thank you. Of course, I'll add the references and corrections as soon as I can.

**Comment by JackCrusoe47** [comment]
@jorgepiloto I'm sorry but how do I run `tox -e reformat && tox -e check`? Currently, I download the repository each time and try to check changes with my PC without using git.  Is this something I should run directly on GitHub?

I apologize, I am still figuring out how to use git and GitHub.

**Comment by jorgepiloto** [comment]
It looks like the quality checks are passing now. Did you finally managed how to use `tox` in the command line, @JackCrusoe47?

**Comment by JackCrusoe47** [comment]
> It looks like the quality checks are passing now. Did you finally managed how to use `tox` in the command line, @JackCrusoe47?

Well, kind of I think. I downloaded the full repository manually and then installed tox and run the commands. But a lot of the white space errors remained. So I fixed them manually. Is there an automated way to do this?

**Comment by jorgepiloto** [comment]
#### About using git

The workflow of using GitHub and git is described in the [contributing](https://docs.poliastro.space/en/latest/contributing.html#pull-request-workflow) docs.

Essentially you must clone the repository only once. From then one, you can update your cloned repository (local repo) with new changes which got merged in this online repository (upstream repo).

To update your local main branch to the upstream one you can run:

```
# Make sure you are in the main branch of the project
git checkout main

# Fetch upstream changes and update your local main branch
git fetch upstream main && git rebase upstream/main
```

You will get used to the git workflow soon or later. I would recommend you using a graphical git client and slowly move on towards the using of git via the command line 👍🏽


#### Quality checks and white spaces at EOL

Normally, running `tox -e reformat` will apply most of the required changes. However, some others require a manual changes (unused imports, variables...). 

Regarding white spaces at the end of the line, this depends on the text editor or IDE you are  using and how it is configured. I am pretty sure there is a plugin or extension which automatically removes white spaces or shows those every time you save the file you are editing.

**Review by jorgepiloto — APPROVED** [review]
Everything looking good from my side, approving this 👍🏽

**Comment by JackCrusoe47** [comment]
@jorgepiloto Thank you, Jorge. I will try to clone the repo with git and try to work with git workflow as you suggested. As for IDE, I work with a bit of notepad++ and spyder.

**Review by astrojuanlu — APPROVED** [review]
Thanks a lot for this contribution @JackCrusoe47 ! The code looks very good and the plots look awesome 😍 And thanks too to @jorgepiloto for the review 💪🏽 

It's weird that you were observing the quality checks failing, in principle @pre-commit-ci was automatically modifying pull requests. I'll look around and see if it still working or not.

Regarding the coverage, you're right @jorgepiloto that we should not include `contrib/` in the measurements. I opened #1452 to track that, no need to block this pull request because of it.

**Comment by astrojuanlu** [comment]
pre-commit.ci run

**Comment by JackCrusoe47** [comment]
Thank you @astrojuanlu for merging. I hope to add further features to the library.

**Comment by astrojuanlu** [comment]
> It's weird that you were observing the quality checks failing, in principle @pre-commit-ci was automatically modifying pull requests. I'll look around and see if it still working or not.

I now fixed it in #1453, there was a misconfiguration.

About the coverage, @jorgepiloto can you check if #1453 fixes the issue? I tried to run the coverage locally but the `contrib/` didn't show up in the reports, not sure where you were spotting it.

**Pull request merged**
